### ISSUE TITLE
ArgoCD/ArgoRollouts integration

### ISSUE THREAD

**Issue by ilya-lesikov** [werf/werf#4115]
## Argo features that we want

ArgoCD
* Pull-model
* Rich GUI
* Questionable:
  - Multicluster delivery (possible in Push-model with a CI/build server like GitLab CI)
  - Cold cluster: backup cluster that syncs with the main one
  - Self-healing (unreliable: if the manifest field is not set in rendered manifest then Argo won't track any changes to this field)

Argo Rollouts:
* Advanced deployment strategies: canary, blue-green

## What werf brings to the Argo users

- consistent and effective Continuous Delivery (build, local development, deployments to review environments)
- better observability and failing fast (kubedog)

## How to integrate

- Render manifests and pass it to the Argo deployment controller (easy)
- Publish chart with images (easy)
- werf bundle integration (hard)
- kubedog integration (hard)

## Roadmap

- Make a complicated deployment demo with ArgoCD/ArgoRollouts
- Implement easy integrations with werf
- Decide whether we want further integration

**Comment by ilya-lesikov** [comment]
### TL;DR: new command `werf bundle render` renders K8s manifests from bundle for further deployment with other CD tools (ArgoCD?) or debugging.

## New command: `werf bundle render`

Since v1.2.67 a new command `werf bundle render` can be used to render Kubernetes manifests from a published werf bundle.

Resulting manifests can be passed to other CD tools, which handle the deployment part, if you can't use werf for deployment and required to use something else.

## Difference with `werf render`

`werf bundle render`, unlike `werf render`, *does not require access to the git repository or werf configuration files*. It requires access either to the OCI-repository with a werf bundle or to the directory with a downloaded and extracted bundle.

Thus you should use `werf render` to integrate with push-based CD tools (Skaffold), but use `werf bundle render` to integrate werf with pull-based (GitOps) CD tools (ArgoCD).

## Example

Execute in CI in the app git repository:
```shell
$ werf bundle publish --repo registry.mydomain.io/project
```

Execute on GitOps/pull-server, which will handle the deployment part and does not have app git repository access:
```shell
$ werf bundle render --repo registry.mydomain.io/project --release myproject --namespace myproject --output manifests.yaml
$ <my-gitops-utility> deploy -f manifests.yaml
```

**Comment by shurup** [comment]
[This article](https://blog.werf.io/new-werf-mode-combining-werf-argo-cd-into-a-unified-ci-cd-process-d49bce7f3be1) explains how you can use Argo CD with werf bundles.

**Comment by ilya-lesikov** [comment]
The integration with ArgoCD is mostly completed at this point, although it has some serious limitations. We are thinking about new integration of werf with ArgoCD, done in a different way: #6158

Argo Rollouts should work as they are with werf, no extra setup needed. But we can add tracking for Rollout Custom Resources, which is now missing: #6157
